File: Server (ASP.NET Core)/Petadmin/Models/AnimalPage.cs
using System.Collections.Generic;
using Petadmin.Core.Models;
using PetAdmin.Core.Models;

namespace Petadmin.Models
{
    public class AnimalPage
    {
        public Animal Animal { get; set; }
        public Owner AnimalOwner { get; set; }
        public List<Visit> VisitsList { get; set; }
        public AnimalPageLists Lists { get; set; }
        public List<PreventiveReminder> preventiveRemindersList { get; set; }
        public List<string> RemindersList { get; set; }
    }
}

File: Client (Angular)/models/animal-page.ts
import {Animal, AnimalEntity} from './animal';
import {Visit} from './visit';
import {Owner, OwnerEntity} from './owner';
import {AnimalPageLists, AnimalPageListsEntity} from './animal-page-lists';
import {PreventiveReminder} from './preventive-reminder';
import {Reminder} from './reminder';

export interface AnimalPage {
  animal: Animal;
  animalOwner: Owner;
  visitsList: Visit[];
  lists: AnimalPageLists;
  preventiveRemindersList: PreventiveReminder[];
  remindersList: Reminder[];
}

export class AnimalPageEntity implements AnimalPage{
  constructor() {
    this.animal = new AnimalEntity();
    this.animalOwner = new OwnerEntity();
    this.lists = new AnimalPageListsEntity();
    this.preventiveRemindersList = [];
    this.visitsList = [];
    this.remindersList = [];
  }
  animal: Animal;
  animalOwner: Owner;
  lists: AnimalPageLists;
  preventiveRemindersList: PreventiveReminder[];
  visitsList: Visit[];
  remindersList: Reminder[];
}
